android/jni/tun2socks/tun2socks.c

```diff
@@ -1153,10 +1164,23 @@ int process_arguments (void)
             return 0;
         }
     }
-    // resolve local DNS server address
-    if (options.dns_server_address) {
-        if (!BAddr_Parse2(&dns_server_address, options.dns_server_address, NULL, 0, 0)) {
-            BLog(BLOG_ERROR, "local dns server addr: BAddr_Parse2 failed");
+
+    if (options.transparent_dns) {
+        if (!options.dns_resolver_address) {
+            BLog(BLOG_ERROR,
+                 "transparentDNS requires a DNS resolver address");
+            return 0;
```

Comment:
Again, we can open a issue for this since it is not immediately related to the current PR and instead (likely) affects the whole project.